lib/orbit-common/memory-source.js

```diff
@@ -198,10 +109,84 @@ var MemorySource = Source.extend({
     });
   },
 
+  _findLinked: function(type, id, link, options) {
+    var _this = this;
+
+    if (options) throw new Exception('`MemorySource#findLinked` does not support `options` argument');
+
+    return new Orbit.Promise(function(resolve) {
+      var result = _this._fetchLinked(type, id, link);
+
+      resolve(result);
+    });
+  },
+
   /////////////////////////////////////////////////////////////////////////////
   // Internals
   /////////////////////////////////////////////////////////////////////////////
 
+  _fetchAll: function(type) {
+    var records = [];
+    var dataForType = this.retrieve([type]);
+
+    if (!dataForType) throw new RecordNotFoundException(type);
+
+    for (var i in dataForType) {
+      if (dataForType.hasOwnProperty(i)) {
+        records.push( dataForType[i] );
+      }
+    }
+
+    return records;
+  },
+
+  _fetchMany: function(type, ids) {
+    var _this = this;
+    var records = [];
+    var notFound = [];
+    var id;
+    var record;
+
+    for (var i = 0, l = ids.length; i < l; i++) {
+      id = _this.getId(type, ids[i]);
```

Comment:
does this only work with primary key, or does it also abstract secondary key finds?